[4.0] Add publish up/down information to articles list

bembelimen opened the pull request:
Pull Request for Issue #21930 .

### Summary of Changes
Add tooltip information for publish_up/down information + code improvements


### Testing Instructions
Set publish up/down dates, hover the icons in the stage column.


### Expected result
Popover information appear.


### Actual result
No popover information

bembelimen committed:
Add publish up/down information to articles list

infograf768 commented on administrator/language/en-GB/en-GB.com_content.ini:
```diff
@@ -14,6 +14,7 @@ COM_CONTENT_ATTRIBS_FIELDSET_LABEL="Options"
 COM_CONTENT_BATCH_OPTIONS="Batch process the selected articles"
 COM_CONTENT_BATCH_TIP="If a category is selected for move/copy, any actions selected will be applied to the copied or moved articles. Otherwise, all actions are applied to the selected articles."
 COM_CONTENT_CHANGE_ARTICLE="Select or Change article"
+COM_CONTENT_CHANGE_STAGE="Change stage"
```
Alpha order

bembelimen committed:
Change ordering

brianteeman commented:
I have tested this item :red_circle: unsuccessfully on 8cc3ae8f61483766dd1b952dc981385606fa1a2d<br /><br />i thought it was working great and was about to post thank you etc when i decided to see what happens when you change the state. With this pr when yoyu try to change the state of any article you get an error message<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21942">issues.joomla.org/tracker/joomla-cms/21942</a>.</sub>

bembelimen committed:
Moved default values to the layout (for override purposes)
Correct the margins/paddings of inactive icons
Fix error message when executing a transition

bembelimen commented:
Thanks @brianteeman

Quy commented on layouts/joomla/button/transition-button.php:
```diff
@@ -0,0 +1,73 @@
+$tipTitle = $options['tip_title'];
+$checkboxName = $options['checkbox_name'];
+?>
+<?php if($only_icon): ?>
```
Add space after `if` and before `:`

Quy commented on layouts/joomla/button/transition-button.php:
```diff
@@ -0,0 +1,73 @@
+?>
+<?php if($only_icon): ?>
+	<span class="tbody-icon mr-1">
+		<span class="<?php echo $this->escape($icon ?? ''); ?> <?php echo $tip ? 'hasPopover' : '' ?>"
```
Add semicolon and repeat below.

Quy commented on layouts/joomla/button/transition-button.php:
```diff
@@ -0,0 +1,73 @@
+	<div class="mr-auto"><?php echo $this->escape($options['stage']); ?></div>
+<?php else: ?>
+	<a class="tbody-icon mr-1 data-state-<?php echo $this->escape($value ?? ''); ?> <?php echo $this->escape(!empty($disabled) ? 'disabled' : null); ?> <?php echo $tip ? 'hasPopover' : '' ?>"
+		<?php if (empty($disabled)): ?>
```
Add space before `:`

Quy commented on layouts/joomla/button/transition-button.php:
```diff
@@ -0,0 +1,73 @@
+			$transitions = array_merge($default, $options['transitions']);
+
+			$attribs = [
+				'id'	=> 'transition-select_' . (int) $id,
```
Remove extra spaces before `=>` and repeat below.

bembelimen committed:
Fix sniffer issues

bembelimen committed:
Align equal sign

Quy commented on layouts/joomla/button/transition-button.php:
```diff
@@ -0,0 +1,73 @@
+	</span>
+	<div class="mr-auto"><?php echo $this->escape($options['stage']); ?></div>
+<?php else: ?>
+	<a class="tbody-icon mr-1 data-state-<?php echo $this->escape($value ?? ''); ?> <?php echo $this->escape(!empty($disabled) ? 'disabled' : null); ?> <?php echo $tip ? 'hasPopover' : '' ?>"
```
Add semicolon

Quy commented on administrator/components/com_content/tmpl/articles/default.php:
```diff
@@ -200,42 +191,28 @@
+													->addState(ContentComponent::CONDITION_TRASHED, '', 'trash', 'COM_CONTENT_CHANGE_STAGE', ['tip_title' => 'JTRASHED'])
+													->setLayout('joomla.button.transition-button')
+													->render($item->stage_condition, $i, $options, $item->publish_up, $item->publish_down);
+
```
Remove extra blank lines

Quy commented on layouts/joomla/button/transition-button.php:
```diff
@@ -0,0 +1,73 @@
+		></span>
+	</span>
+	<div class="mr-auto"><?php echo $this->escape($options['stage']); ?></div>
+<?php else: ?>
```
Add space before `:`

Quy commented on layouts/joomla/button/transition-button.php:
```diff
@@ -0,0 +1,73 @@
+			$transitions = array_merge($default, $options['transitions']);
+
+			$attribs = [
+				'id'		=> 'transition-select_' . (int) $id,
```
Remove extra spaces before `=>` and repeat below.

brianteeman commented:
I have tested this item :white_check_mark: successfully on f179785cda37c0620bdcabb2cf17b2dc15161b50<br /><br />That works now - thanks - one less issue 

Obviously the cs issues need to be fixed but then its good to go<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21942">issues.joomla.org/tracker/joomla-cms/21942</a>.</sub>

bembelimen committed:
Fix sniffer issues

bembelimen committed:
Tabs => Spaces

ghazal commented:
I have tested this item :white_check_mark: successfully on 1a507bc656df8fd3c64550f7dad1b63847bc73c3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21942">issues.joomla.org/tracker/joomla-cms/21942</a>.</sub>

franz-wohlkoenig commented:
Ready to Commit after two successful tests.
